### PR TITLE
optimize:use map instead of if else judge for more change in the future

### PR DESCRIPTION
**Pull request by coconut4j**
### Ⅰ. Describe what this PR did
use two ImmutableMap instead of lots of if else judge, and the code is more readability,more convenient to modify.

### Ⅴ. Special notes for reviews
 if enum class don't use  Strategy Pattern,you can use map to optimize . can also use  the same  to io.seata.server.storage.db.store.DataBaseTransactionStoreManager#writeSession for db.

### CODE REVIEW

**Comment by xilou31** [comment]
I think it's a bit over-engineered, if you want to optimize lots of "if else", you can use switch.

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 1]
```diff
@@ -92,20 +97,58 @@ public static RedisTransactionStoreManager getInstance() {
...
+     */
+    public static volatile ImmutableMap<LogOperation, Function<BranchTransactionDO, Boolean>> branchMap;
+
+    /**
+     * init globalMap
+     *
+     * @param
+     * @return void
+     */
+    @PostConstruct
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:117]: 注解应该是无效的，这个manager不是一个spring bean，应该放到getInstance 进行初始化

**Comment by jsbxyyx** [comment]
git name and email relation github id.
the init xxx map code move to class constructor.

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 2]
```diff
@@ -92,20 +97,58 @@ public static RedisTransactionStoreManager getInstance() {
         return instance;
     }
 
+
+    /*
+    Map for LogOperation Global Operation
+    */
+    public static volatile ImmutableMap<LogOperation, Function<GlobalTransactionDO, Boolean>> globalMap;
```
Comment (l81893521) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:104]: Why not use one map instead of two?

Comment (coconut4j) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:104]: in this class，two map Fuction<T,R> has diff T

**Comment by caohdgege** [comment]
plz resolve the code conflicts, or maybe you can create a new branch from develop & resubmit the pr.

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java (hunk 2 above)
Comment (l81893521) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:104]: ok, please fix another review.

**Pull request closed without merging**